node/src/components/deploy_acceptor/tests.rs

```diff
@@ -0,0 +1,584 @@
+            // Check that repeated valid deploys raise the `PutToStorageResult` with the
+            // `is_new` flag as false.
+            TestScenario::FromClientRepeatedValidDeploy
+            | TestScenario::FromPeerRepeatedValidDeploy => {
+                matches!(
+                    event,
+                    Event::DeployAcceptor(super::Event::PutToStorageResult { is_new: false, .. })
+                )
+            }
+            // Deploys received from a peer should not raise `ContractRuntimeRequest` events
+            // as part of the validation process.
+            // This test scenario should therefore, keep running and eventually timeout.
+            TestScenario::FromPeerRegression => {
+                matches!(
+                    event,
+                    Event::ContractRuntime(ContractRuntimeRequest::Query { .. })
+                )
+            }
+        }
+    };
+    runner
+        .reactor_mut()
+        .set_condition_checker(Box::new(stopping_condition));
+
+    loop {
+        if runner.try_crank(&mut rng).await.is_some() {
+            if runner.reactor().condition_result() {
+                break;
+            }
+        } else {
+            time::sleep(POLL_INTERVAL).await;
+        }
+    }
+
+    {
+        // Assert that the deploy is present in the case of a valid deploy.
+        // Conversely, assert its absence in the invalid case.
+        let is_in_storage = runner
+            .reactor()
+            .inner()
+            .storage
+            .get_deploy_by_hash(*deploy.id())
+            .is_some();
+
+        if test_scenario.is_valid_deploy_case() {
+            assert!(is_in_storage)
+        } else {
+            assert!(!is_in_storage)
+        }
+    }
+
+    deploy_receiver.await.unwrap()
+}
+
+async fn run_deploy_acceptor(test_scenario: TestScenario) -> Result<(), super::Error> {
+    time::timeout(TIMEOUT, run_deploy_acceptor_without_timeout(test_scenario))
+        .await
+        .unwrap()
+}
+
+#[tokio::test]
+async fn should_accept_valid_deploy_from_peer() {
+    let result = run_deploy_acceptor(TestScenario::FromPeerValidDeploy).await;
+    assert!(result.is_ok())
+}
+
+#[tokio::test]
+async fn should_reject_invalid_deploy_from_peer() {
+    let result = run_deploy_acceptor(TestScenario::FromPeerInvalidDeploy).await;
+    assert!(matches!(result, Err(super::Error::InvalidDeploy(_))))
+}
+
+#[tokio::test]
+async fn should_accept_valid_deploy_from_client() {
+    let result = run_deploy_acceptor(TestScenario::FromClientValidDeploy).await;
+    assert!(result.is_ok())
+}
+
+#[tokio::test]
+async fn should_reject_invalid_deploy_from_client() {
+    let result = run_deploy_acceptor(TestScenario::FromClientInvalidDeploy).await;
+    assert!(matches!(result, Err(super::Error::InvalidDeploy(_))))
+}
+
+#[tokio::test]
+async fn should_reject_valid_deploy_from_client_for_invalid_account() {
+    let result = run_deploy_acceptor(TestScenario::FromClientMissingAccount).await;
```

Comment:
Minor nit, consider using either `invalid` or `missing` word, not both.

Reply:
Applied suggestion